### PR TITLE
Preloading

### PR DESCRIPTION
**Pull request by vadim11041996**
Added preloader

### CODE REVIEW

File: src/app/modules/teachers/containers/teachers-page/teachers-page.component.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class TeachersPageComponent implements OnInit, OnDestroy {
   constructor(private teacherService: TeachersService,
     private modalService: BsModalService,
     private route: ActivatedRoute,
-    private router: Router) { }
+    public router: Router) { }
```
Comment (alexandrubereghici) [inline src/app/modules/teachers/containers/teachers-page/teachers-page.component.ts]: why?

File: src/app/modules/teachers/containers/teachers-page/teachers-page.component.html [hunk 2]
```diff
@@ -39,5 +36,4 @@ <h2>Want to add a new teacher ? </h2>
                 [(ngModel)]="currentPage"
```
Comment (crischiucucu) [inline src/app/modules/teachers/containers/teachers-page/teachers-page.component.html]: If you specify  [maxSize]="limit" you don't need  [maxSize]="limit"

File: src/app/modules/teachers/containers/teachers-page/teachers-page.component.ts [hunk 3]
```diff
@@ -47,8 +47,13 @@ export class TeachersPageComponent implements OnInit, OnDestroy {
     });
     this.teacherService.getTeachers(this.offset, this.limit).pipe(takeUntil(this.destroy$))
       .subscribe(teachers => this.teachers = teachers);
-    this.teacherService.getNumberOfTeachers().pipe(takeUntil(this.destroy$))
-      .subscribe(teachers => this.totalItems = teachers);
+    this.teacherService.getNumberOfTeachers()
```
Comment (crischiucucu) [inline src/app/modules/teachers/containers/teachers-page/teachers-page.component.ts]: You could use crud-service here

File: src/app/modules/teachers/containers/teachers-page/teachers-page.component.ts [hunk 4]
```diff
@@ -47,8 +47,13 @@ export class TeachersPageComponent implements OnInit, OnDestroy {
...
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(teachers => {
+        this.totalItems = teachers;
+      }, err => {
+        console.log(err);
```
Comment (crischiucucu) [inline src/app/modules/teachers/containers/teachers-page/teachers-page.component.ts]: Errors should be displayed into the user interface

**Pull request merged**